builtin/plugins/chainconfig/chainconfig.go

```diff
@@ -407,6 +408,23 @@ func HarvestPendingActions(ctx contract.Context, buildNumber uint64) ([]*Action,
 	return actions, nil
 }
 
+func CheckFeaturesBuildNumber(ctx contract.Context, blockHeight, buildNumber uint64) error {
+	featureRange := ctx.Range([]byte(featurePrefix))
+	for _, m := range featureRange {
+		var f Feature
+		if err := proto.Unmarshal(m.Value, &f); err != nil {
+			return errors.Wrapf(err, "failed to unmarshal feature %s", string(m.Key))
+		}
+		if f.Status != FeatureEnabled {
+			continue
+		}
+		if f.BuildNumber > buildNumber && f.BlockHeight < blockHeight {
```

Comment:
We don't need to check block height. Just do
```
if f.BuildNumber > buildNumber && f.Status == FeatureEnabled
```
We should also error out the minimum build number that can run on this data